lib/connection_pool.rb

```diff
@@ -44,6 +44,32 @@ def self.wrap(options, &block)
     Wrapper.new(options, &block)
   end
 
+  def self.after_fork
+    ObjectSpace.each_object(ConnectionPool) do |pool|
+      # We're on after fork, so we know all other threads are dead.
+      # All we need to do is to ensure the main thread doesn't have a
+      # checked out connection
+      pool.checkin(force: true)
+
+      # Unfortunately we don't know what method to call to close the connection.
+      # So we simply drop the reference and let Ruby close the sockets.
+      pool.reload { |_c| }
+    end
+  end
+
+  if ::Process.respond_to?(:_fork) # MRI 3.1+
```

Comment:
Right, I think it makes a lot of sense for JRuby & TruffleRuby to either not define _fork or define it as "undefined" which means it raises when called + `Process.respond_to?(:_fork)` is always false.
So the code as-is already works fine (i.e. noops) on JRuby & TruffleRuby.

For the WeakMap version it needs some small tweaks, I'll review that if you push the commit in this PR.

Reply:
> For the WeakMap version it needs some small tweaks, I'll review that if you push the commit in this PR.

You can comment on the commit, I'd be interested in knowing what teaks are needed regardless of whether Mike agrees to include it.

Reply:
I think I'm ok with WeakMap. Added 50ms to the cost of fork is tough to justify.

Reply:
Done.